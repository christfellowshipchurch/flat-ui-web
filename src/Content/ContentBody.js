import React from 'react';
import { Parser } from 'html-to-react';

const htmlToReactParser = new Parser();

const ContentBody = ({ children }) =>
  <div className="my-3">
    {htmlToReactParser.parse(children)}
  </div>

const defaultProps = {};
const propTypes = {};

ContentBody.defaultProps = defaultProps;
ContentBody.propTypes = propTypes;

export default ContentBody;
