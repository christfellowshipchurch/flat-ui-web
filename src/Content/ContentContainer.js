import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  contentContainer, contentMedia, contentRight, contentLeft, contentTop, contentBottom
} from '../css/styles.module.css'

import * as Media from '../Media'

const ContentContainer = ({
  layout, imageUrl, imageAlt, videoUrl, ratio, children, className
}) => {
  const layouts = {
    default: contentBottom,
    inverted: contentTop,
    left: contentLeft,
    right: contentRight,
  };

  return (
    <div className={classNames(contentContainer, className)}>
      {imageUrl || videoUrl
        ? (
          <div className={contentMedia}>
            <Media.Item ratio={ratio} imageUrl={imageUrl} imageAlt={imageAlt} videoUrl={videoUrl} />
          </div>
        )
        : null}
      <div className={layouts[layout]}>
        {children}
      </div>
    </div>
  );
};

const defaultProps = {
  layout: 'default',
  imageUrl: null,
  imageAlt: null,
  videoUrl: null,
  ratio: '16by9',
  className: ''
};

const propTypes = {
  layout: PropTypes.string,
  imageUrl: PropTypes.string,
  imageAlt: PropTypes.string,
  videoUrl: PropTypes.string,
  ratio: PropTypes.string,
  className: PropTypes.string
};

ContentContainer.defaultProps = defaultProps;
ContentContainer.propTypes = propTypes;

export default ContentContainer;
