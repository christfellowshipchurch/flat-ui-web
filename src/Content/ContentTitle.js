import React from 'react';

const ContentTitle = ({ children }) =>
  <h2>
    {children}
  </h2>

const defaultProps = {};
const propTypes = {};

ContentTitle.defaultProps = defaultProps;
ContentTitle.propTypes = propTypes;

export default ContentTitle;
