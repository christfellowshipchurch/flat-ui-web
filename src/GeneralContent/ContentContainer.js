import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  contentContainer, contentMedia, contentRight, contentLeft, contentTop, contentBottom
} from '../css/styles.module.css'

import * as Media from '../Media'

const ContentContainer = ({
  layout, imageUrl, imageAlt, videoUrl, ratio, children, className
}) => {
  const layouts = {
    default: contentBottom,
    inverted: contentTop,
    left: contentLeft,
    right: contentRight,
  };

  return (
    <div className={classNames(contentContainer, className)}>
      <div className={contentMedia}>
        <Media.Item ratio={ratio} imageUrl={imageUrl} imageAlt={imageAlt} videoUrl={videoUrl} />
      </div>
      <div className={layouts[layout]}>
        {children}
      </div>
    </div>
  );
};

const defaultProps = {
  layout: 'default',
  imageUrl: '',
  imageAlt: '',
  videoUrl: null,
  ratio: '16by9',
  className: ''
};

const propTypes = {
  layout: PropTypes.string,
  imageUrl: PropTypes.string,
  imageAlt: PropTypes.string,
  videoUrl: PropTypes.string,
  ratio: PropTypes.string,
  className: PropTypes.string
};

ContentContainer.defaultProps = defaultProps;
ContentContainer.propTypes = propTypes;

export default ContentContainer;
