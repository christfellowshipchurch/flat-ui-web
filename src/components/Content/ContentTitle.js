import React from 'react'

const ContentTitle = ({ className, children, larger = false }) => (
  larger ?
    <h1 className={className}>
      {children}
    </h1> :
    <h2 className={className}>
      {children}
    </h2>
)

const defaultProps = {};
const propTypes = {};

ContentTitle.defaultProps = defaultProps;
ContentTitle.propTypes = propTypes;

export default ContentTitle;
