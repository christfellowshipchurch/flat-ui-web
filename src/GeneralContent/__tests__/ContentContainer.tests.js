import React from 'react'
import ContentContainer from '../ContentContainer'
import ContentTitle from '../ContentTitle'
import ContentBody from '../ContentBody'
import renderer from 'react-test-renderer'

describe('ContentContainer', () => {
    it('renders an image and content in default layout', () => {
        const tree = renderer.create(
            <ContentContainer
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="default"
            >
                <ContentTitle>Title</ContentTitle>
                <ContentBody>Content body.</ContentBody>
            </ContentContainer>
        )

        expect(tree).toMatchSnapshot();
    })
})