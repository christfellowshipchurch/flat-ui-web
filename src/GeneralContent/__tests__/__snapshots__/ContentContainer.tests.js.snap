// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer renders an image and content in default layout 1`] = `
<div
  className=""
>
  <div>
    <div
      className=""
    >
      <div>
        <div />
      </div>
    </div>
  </div>
  <div>
    <h2>
      Title
    </h2>
    <div
      className="my-3"
    >
      Content body.
    </div>
  </div>
</div>
`;
